File: src/main/java/com/claudio_personagens/bookingcharactersapi/base/AbstractBaseEntity.java
package com.claudio_personagens.bookingcharactersapi.base;

import lombok.Data;

import javax.persistence.Column;
import javax.persistence.Id;
import javax.persistence.MappedSuperclass;
import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.UUID;

@MappedSuperclass
@Data
public abstract class AbstractBaseEntity implements Serializable {

    private static final long serialVersionUID = 1L;

    @Id
    @Column(length = 36)
    private String id;
    private LocalDateTime createdAt;

    public AbstractBaseEntity() {
        this.id = UUID.randomUUID().toString();
        this.createdAt = LocalDateTime.now();
    }

    @Override
    public int hashCode() {
        return id.hashCode();
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj)
            return true;
        if (obj == null)
            return false;
        if (!(obj instanceof AbstractBaseEntity)) {
            return false;
        }
        AbstractBaseEntity other = (AbstractBaseEntity) obj;
        return getId().equals(other.getId());
    }
}

File: src/main/java/com/claudio_personagens/bookingcharactersapi/base/AbstractBaseEntity.kt
package com.claudio_personagens.bookingcharactersapi.base

import java.time.Instant
import java.util.UUID
import javax.persistence.Column
import javax.persistence.Id
import javax.persistence.MappedSuperclass

@MappedSuperclass
abstract class AbstractBaseEntity (
    @Id
    @Column(length = 36)
    open val id: UUID = UUID.randomUUID(),
    open var createdAt: Instant = Instant.now()
)